packages/kbn-router-to-openapispec/src/generate_oas.test.ts

```diff
@@ -126,6 +161,70 @@ describe('generateOpenApiDocument', () => {
     });
   });
 
+  describe('Zod', () => {
+    it('generates the expected OpenAPI document for the shared schema', () => {
+      const [routers, versionedRouters] = createTestRoutersForZod({
+        routers: { testRouter: { routes: [{ method: 'get' }, { method: 'post' }] } },
+        versionedRouters: { testVersionedRouter: { routes: [{}] } },
+        bodySchema: createZodSharedSchema(),
+      });
+      expect(
+        generateOpenApiDocument(
+          {
+            routers,
+            versionedRouters,
+          },
+          {
+            title: 'test',
+            baseUrl: 'https://test.oas',
+            version: '99.99.99',
+          }
+        )
+      ).toMatchObject(sharedOas);
+    });
+
+    it('handles recursive schemas', () => {
+      const recursiveSchema: z.ZodType<ZodRecursiveType> = baseCategorySchema.extend({
```

Comment:
@jloleysens I've added this test but wasn't sure what I should expect to see in the related snapshot. Would you please let me know if this test is meaningful for Zod?

Btw, in the console, I got the following warning for this test:

![image](https://github.com/elastic/kibana/assets/12370520/637e517f-f9b9-49b8-a059-c62e26842c69)

Reply:
That's good to know. See my comment in the snapshot.